### PR TITLE
Improve new menu field

### PR DESCRIPTION
**Pull request by bembelimen**
### Summary of Changes
Improve parameter loading to have real default values.

### Testing Instructions
It's a bit hard to test. You have to load the com_menus/models/fields/modal/menu.php field in your xml or took an existing one like: https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_menus/models/forms/item_alias.xml

After that you have to play with the parameters:
```
select="true"
new="true"
edit="true"
clear="true"
```

at the end all the old combinations should still work + if you remove this four parameters it should be the same as:
```
select="true"
new="false"
edit="false"
clear="true"
```

### CODE REVIEW

**Comment by infograf768** [comment]
@andrepereiradasilva 
The field was improved in https://github.com/joomla/joomla-cms/pull/11916
can you have a look?

**Comment by coolcat-creations** [comment]
I have tested this item :white_check_mark: successfully on 68c539e2bda9ee9b0b951b127a816f98023f84a2<br /><br />Works like described<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13085">issues.joomla.org/tracker/joomla-cms/13085</a>.</sub>

**Comment by andrepereiradasilva** [comment]
@bembelimen i think this is the right approach, but IMHO we need to do the same for all modal fields. (article, newsfeed, contact, category, and weblink).

Also we should add some ACL check at least for core.create. 

But of course this would be another PRs,

**Comment by bembelimen** [comment]
@andrepereiradasilva Yes, you're right, that would be the next step I think.

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 68c539e2bda9ee9b0b951b127a816f98023f84a2<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13085">issues.joomla.org/tracker/joomla-cms/13085</a>.</sub>

File: administrator/components/com_menus/models/fields/modal/menu.php [hunk 1]
```diff
@@ -24,6 +24,116 @@ class JFormFieldModal_Menu extends JFormField
...
+	protected $allowEdit = false;
+
+	/**
+	 * Method to get certain otherwise inaccessible properties from the form field object.
+	 *
+	 * @param   string  $name  The property name for which to the the value.
+	 *
+	 * @return  mixed  The property value or null.
+	 *
+	 * @since   3.2
```
Comment (andrepereiradasilva) [inline administrator/components/com_menus/models/fields/modal/menu.php]: please use `__DEPLOY__VERSION__`

File: administrator/components/com_menus/models/fields/modal/menu.php [hunk 2]
```diff
@@ -24,6 +24,116 @@ class JFormFieldModal_Menu extends JFormField
...
+
+	/**
+	 * Method to set certain otherwise inaccessible properties of the form field object.
+	 *
+	 * @param   string  $name   The property name for which to the the value.
+	 * @param   mixed   $value  The value of the property.
+	 *
+	 * @return  void
+	 *
+	 * @since   3.2
```
Comment (andrepereiradasilva) [inline administrator/components/com_menus/models/fields/modal/menu.php]: please use `__DEPLOY__VERSION__`

**Comment by bembelimen** [comment]
Thanks @andrepereiradasilva

**Pull request merged**